engine/src/main/java/io/zeebe/engine/nwe/container/ProcessProcessor.java

```diff
@@ -22,69 +26,173 @@
+        .ifRightOrLeft(
+            ok -> stateTransitionBehavior.transitionToActivated(context),
+            failure -> incidentBehavior.createIncident(failure, context));
   }
 
   @Override
-  public void onChildTerminated(
-      final ExecutableFlowElementContainer element,
-      final BpmnElementContext flowScopeContext,
-      final BpmnElementContext childContext) {
+  public void onActivated(
+      final ExecutableFlowElementContainer element, final BpmnElementContext context) {
 
-    if (flowScopeContext.getIntent() == WorkflowInstanceIntent.ELEMENT_TERMINATING
-        && stateBehavior.isLastActiveExecutionPathInScope(childContext)) {
-      stateTransitionBehavior.transitionToTerminated(flowScopeContext);
+    boolean triggeredByEvent = false;
+    if (element.hasMessageStartEvent() || element.hasTimerStartEvent()) {
 
-    } else {
-      eventSubscriptionBehavior.publishTriggeredEventSubProcess(flowScopeContext);
+      triggeredByEvent = eventSubscriptionBehavior.publishTriggeredStartEvent(context);
     }
-  }
 
-  @Override
-  public Class<ExecutableFlowElementContainer> getType() {
-    return ExecutableFlowElementContainer.class;
-  }
+    if (!triggeredByEvent) {
 
-  @Override
-  public void onActivating(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {}
+      final var noneStartEvent = element.getNoneStartEvent();
+      if (noneStartEvent == null) {
+        throw new BpmnProcessingException(
+            context, "Expected to activate the none start event of the process but not found.");
+      }
 
-  @Override
-  public void onActivated(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {}
+      stateTransitionBehavior.activateChildInstance(context, noneStartEvent);
+    }
+  }
 
   @Override
   public void onCompleting(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {}
+      final ExecutableFlowElementContainer element, final BpmnElementContext context) {
+
+    eventSubscriptionBehavior.unsubscribeFromEvents(context);
+    stateTransitionBehavior.transitionToCompleted(context);
+  }
 
   @Override
   public void onCompleted(
-      final ExecutableFlowElementContainer element, final BpmnElementContext context) {}
+      final ExecutableFlowElementContainer element, final BpmnElementContext context) {
+
+    final var parentWorkflowInstanceKey = context.getParentWorkflowInstanceKey();
+    final var parentElementInstanceKey = context.getParentElementInstanceKey();
+
+    if (parentWorkflowInstanceKey > 0) {
+      // workflow instance is created by a call activity
+
+      // TODO (saig0): move responsibility to call activity (#4473)
+      final var parentElementInstance = stateBehavior.getParentElementInstanceContext(context);
+
+      if (parentElementInstance != null
+          && parentElementInstance.getIntent() == WorkflowInstanceIntent.ELEMENT_ACTIVATED) {
```

Comment:
Should we maybe log a warning or throw an exception when parent is not `ACTIVATED`?

Reply:
No, I don't think so. I can happen that the call activity is in state `terminating` or `terminated` (e.g. if a boundary event is triggered).